services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -250,16 +260,132 @@ class MetersSpec extends BaseSpecification {
+                assert meterEntry.burstSize == getExpectedBurst(switchMeterEntries.switchId, newBandwidth)
+                //id of the meter is the same
+                assert originalMeters[switchMeterEntries.switchId]
+                        .findAll(flowMeters)*.meterId.contains(meterEntry.meterId)
+            }
+        }
+
+        and: "Non-default meter rate and burst are actually changed to expected values both on src and dst switch"
+        [src.dpId, dst.dpId].each { SwitchId swId ->
+            northbound.getAllMeters(swId).meterEntries.findAll(flowMeters)
+                .each { meterEntry ->
+                    assert meterEntry.rate == newBandwidth
+                    assert meterEntry.burstSize == getExpectedBurst(swId, newBandwidth)
+                    //id of the meter is the same
+                    assert originalMeters[swId].findAll(flowMeters)*.meterId.contains(meterEntry.meterId)
+                }
+        }
+
+        and: "Default meters are unchanged"
+        [src.dpId, dst.dpId].each { SwitchId swId ->
+            assert expect(northbound.getAllMeters(swId).meterEntries.findAll(defaultMeters).sort(),
+                    sameBeanAs(originalMeters[swId].findAll(defaultMeters).sort())
+                            .ignoring("timestamp"))
+        }
+
+        and: "Switch rules are unchanged"
+        [src.dpId, dst.dpId].each { SwitchId swId ->
+            assert expect(northbound.getSwitchRules(swId), sameBeanAs(originalRules[swId])
+                    .ignoring("timestamp")
+                    .ignoring("flowEntries.durationNanoSeconds")
+                    .ignoring("flowEntries.durationSeconds")
+                    .ignoring("flowEntries.byteCount")
+                    .ignoring("flowEntries.packetCount"))
+        }
+
+        and: "Cleanup: delete flow"
+        flowHelper.deleteFlow(flow.id)
+
+        where: data << [
+                [
+                        description: "nonCentec-nonCentec",
+                        srcSwitch: nonCentecSwitches[0],
+                        dstSwitch: nonCentecSwitches[1]
+                ],
+                [
+                        description: "centec-centec",
+                        srcSwitch: centecSwitches[0],
+                        dstSwitch: centecSwitches[1]
+                ],
+                [
+                        description: "centec-nonCentec",
+                        srcSwitch: centecSwitches[0],
+                        dstSwitch: nonCentecSwitches[0]
+                ]
+        ]
+    }
+
+    def getExpectedBurst(SwitchId sw, long rate) {
+        def descr = getSwitchDescription(sw)
+        if (descr.contains("NoviFlow")) {
+            return (rate * NOVI_BURST_COEF - 1).setScale(0, RoundingMode.CEILING)
+        } else if (descr.contains("Centec")) {
+            def burst = (rate * burstCoef).toBigDecimal().setScale(0, RoundingMode.FLOOR)
+            if(burst <= CENTEC_MIN_BURST) {
```

Comment:
The bursting logic on Centecs is complicated. Would be great to leave comments here for future reference.